lib/IRGen/IRGenModule.cpp

```diff
@@ -932,6 +937,19 @@ llvm::Constant *swift::getRuntimeFn(llvm::Module &Module,
     fn->addAttributes(llvm::AttributeList::FunctionIndex, buildFnAttr);
     fn->addAttributes(llvm::AttributeList::ReturnIndex, buildRetAttr);
     fn->addParamAttrs(0, buildFirstParamAttr);
+        
+    // Add swiftself and swifterror attributes only when swift_willThrow
+    // swift_willThrow is defined in RuntimeFunctions.def, but due to the
+    // DSL limitation, arguments attributes are not set.
+    // On the other hand, caller of `swift_willThrow` assumes that it's attributed
+    // with `swiftself` and `swifterror`.
+    // This mismatch of attributes would be issue when lowering to WebAssembly.
+    // While lowering, LLVM counts how many dummy params are necessary to match
+    // callee and caller signature. So we need to add them correctly.
+    if (functionName == "swift_willThrow") {
```

Comment:
Could we model this with a flag in the `RuntimeFunctions.def` DSL? The specific function name check feels brittle.

Reply:
This would require some refinement of such DSL, right? Currently it doesn't allow setting attributes on parameters.